x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStep.java

```diff
@@ -44,15 +46,23 @@ public SetSingleNodeAllocateStep(StepKey key, StepKey nextStepKey, Client client
 
     @Override
     public void performAction(IndexMetaData indexMetaData, ClusterState clusterState, ClusterStateObserver observer, Listener listener) {
-        RoutingAllocation allocation = new RoutingAllocation(ALLOCATION_DECIDERS, clusterState.getRoutingNodes(), clusterState, null,
+        final RoutingNodes routingNodes = clusterState.getRoutingNodes();
+        RoutingAllocation allocation = new RoutingAllocation(ALLOCATION_DECIDERS, routingNodes, clusterState, null,
                 System.nanoTime());
         List<String> validNodeIds = new ArrayList<>();
         Optional<ShardRouting> anyShard = clusterState.getRoutingTable().allShards(indexMetaData.getIndex().getName()).stream().findAny();
         if (anyShard.isPresent()) {
+            // Find the highest node version in the cluster - shards can't be allocated to a node running an older version than they're
+            // currently on, so we should only pick one of the newest version nodes to reallocate to
+            Version highestNodeVersion = indexMetaData.getCreationVersion();
+            for (RoutingNode node : routingNodes) {
+                highestNodeVersion = Version.max(highestNodeVersion, node.node().getVersion());
```

Comment:
I think this unnecessarily limits us to the highest node version, rather than any node >= the current version (see comment at the bottom)